src/components/settings/settings-assigned-users/settings-assigned-users.test.js

```diff
@@ -0,0 +1,179 @@
+      alreadyAssignedMessageDisplayed={false}
+      assignedUsers={assignedUsers}
+      credentialsId='credentials-id'
+      hideAlreadyAssignedMessage={() => {}}
+      knowledgeBaseName='Knowledge Base'
+      onDeleteUser={() => {}}
+      onSelectUser={() => {}}
+      requestIsPending={false}
+      {...props}
+    />
+  </Harness>
+);
+
+describe('Given SettingsAssignedUsers', () => {
+  const mockOnDeleteUser = jest.fn();
+  const mockOnSelectUser = jest.fn();
+
+  it('should render SettingsAccessStatusTypes component', () => {
+    const { getByTestId } = renderSettingsAssignedUsers();
+
+    expect(getByTestId('settings-assigned-users')).toBeDefined();
+  });
+
+  it('should display `Assigned users` in the pane header and as the headline', () => {
+    const { getAllByText } = renderSettingsAssignedUsers();
+
+    expect(getAllByText('ui-eholdings.settings.assignedUsers')).toHaveLength(2);
+  });
+
+  describe('when assignedUsers array is empty', () => {
+    it('should display an empty message', () => {
+      const { getByText } = renderSettingsAssignedUsers({
+        assignedUsers: [],
+      });
+
+      expect(getByText('ui-eholdings.settings.assignedUsers.list.emptyMessage')).toBeDefined();
+    });
+  });
+
+  it('should display table column titles', () => {
+    const { getByText } = renderSettingsAssignedUsers();
+
+    expect(getByText('ui-eholdings.settings.assignedUsers.list.name')).toBeDefined();
+    expect(getByText('ui-eholdings.settings.assignedUsers.list.patronGroup')).toBeDefined();
+  });
+
+  it('should display assigned users info', () => {
+    const { getByText } = renderSettingsAssignedUsers();
+
+    expect(getByText('assigned-user-name')).toBeDefined();
+    expect(getByText('assigned-user-patron-group')).toBeDefined();
+  });
+
+  it('should display unassign user icon button', () => {
+    const { getByRole } = renderSettingsAssignedUsers();
+
+    expect(getByRole('button', { name: 'ui-eholdings.settings.assignedUsers.list.unassignUser' })).toBeDefined();
+  });
+
+  describe('when click on unassign user icon button', () => {
+    it('should show a confirmation modal', () => {
+      const {
+        getByRole,
+        getByTestId,
+        getByText,
+      } = renderSettingsAssignedUsers();
+
+      fireEvent.click(getByRole('button', { name: 'ui-eholdings.settings.assignedUsers.list.unassignUser' }));
+
+      expect(getByTestId('unassign-user-confirmation-modal')).toBeDefined();
+
+      const title = 'ui-eholdings.settings.assignedUsers.confirmationModal.title';
+      const message = 'ui-eholdings.settings.assignedUsers.confirmationModal.prompt';
+      const cancelButton = 'ui-eholdings.settings.assignedUsers.confirmationModal.cancelButton';
+      const confirmButton = 'ui-eholdings.settings.assignedUsers.confirmationModal.confirmButton';
+
+      expect(getByText(title)).toBeDefined();
+      expect(getByText(message)).toBeDefined();
+      getByRole('button', { name: cancelButton });
```

Comment:
What is the purpose of these two lines after last `expect`?

Reply:
Hm, I somehow missed this, sorry. Fixed now!